uniffi_bindgen/src/bindings/python/templates/RustBufferStream.py

```diff
@@ -163,6 +163,35 @@ def read{{ canonical_type_name }}(self):
         raise InternalError("Unexpected variant tag for {{ canonical_type_name }}")
         {%- endif %}
 
+    {% when Type::Error with (error_name) -%}
+    {%- let e = ci.get_error_definition(error_name).unwrap().wrapped_enum() %}
+
+    # The Error type {{ error_name }}
+
+    # Top-level read method
+    def read{{ canonical_type_name }}(self):
+        variant = self._unpack_from(4, ">i")
+        try:
+            read_variant_method = getattr(self, 'read{{canonical_type_name}}{}'.format(variant))
+        except AttributeError:
+            raise InternalError("Unexpected variant value for error {{ canonical_type_name }} ({})".format(variant))
+        return read_variant_method()
+
+    # Read methods for each individual variants
+    {%- for variant in e.variants() %}
+
+    def read{{ canonical_type_name }}{{ loop.index }}(self):
```

Comment:
I remain lightly concerned about name conflicts here in pathological cases, e.g. if I had one error named `MyError` and another error named `MyError1`. What do you think about putting the variant helpers into hidden methods with a different prefix like `_readVariant{{ loop.index }}Of{{ canonical_type_name }}` or similar?